systests/rs-sse/rs-sse-base/src/main/java/org/apache/cxf/systest/jaxrs/sse/BookStore.java

```diff
@@ -124,6 +124,7 @@ public void stop() {
             broadcaster.broadcast(createStatsEvent(builder.name("book"), 1000));
             broadcaster.broadcast(createStatsEvent(builder.name("book"), 2000));
 
+            Thread.sleep(2000);
```

Comment:
definitely... I totally agree

Reply:
mmmh... now that I think about it... I guess it would not work because the futures would finish immediately as well, wouldn't they?

Reply:
That's correct, but it shouldn't be an issue because the events are broadcasted through the different endpoints (SSE ones). The call to close the broadcaster will happen eventually, the test case waits some time for data to become available.

Reply:
@csierra I did a minor refactoring of the test cases in question, seem to work fine and respect the events sequence, thanks a lot for finding that.